### PR TITLE
Footer Style Refactor

### PR DESCRIPTION
**Pull request by ThatJoeMoore**
# Work In Progress

# Summary of Changes

**Fixes Issue #210, #234, #255 **

This is a refactor of the footer styles to address issues on sites with lots of floated elements.  It also serves as a general cleanup of the footer styles, and unifies all of the positioning to use Flexbox, instead of inline-block and float styles.

This refactor also adds support for double- and triple-wide columns.

Improves documentation for the ways to use byu-footer.

# Browser Testing

**I have tested these changes in:**

*Add an x in all the boxes that apply. Please mark desktop and mobile
browsers separately.*

## Desktop Browsers

- [x] Google Chrome
- [x] Mozilla Firefox
- [x] Apple Safari
- [x] Microsoft Edge
- [x] Microsoft Internet Explorer 11
- [ ] Other (please specify)

## Mobile Browsers

- [ ] Any browser on iOS
- [ ] Chrome for Android
- [ ] Firefox Mobile for Android
- [ ] Other (please specify)

**We support the last two versions of Chrome, Firefox, Safari, and Edge,
plus Internet Explorer 11.**

### CODE REVIEW

**Comment by ThatJoeMoore** [comment]
@katrialesser Could you try this branch on the calendar site and see if it helps with your issues there?

File: docs/img/galactic-empire.svg [hunk 1]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (kebowdog) [inline docs/img/galactic-empire.svg]: I can't find any reference to these SVGs anywhere in the css, but I'm very excited about their inclusion in the components.

Comment (ThatJoeMoore) [inline docs/img/galactic-empire.svg]: They're referenced in the footer docs (the section about the social media icons).

Comment (ThatJoeMoore) [inline docs/img/galactic-empire.svg]: If you like those, you should also look at the custom footer content demo I added :D.  I may have been feeling a bit cheeky while working on the docs.

**Comment by ThatJoeMoore** [comment]
I'm going to try to retrofit the every-feature page to work with IE. I think I mostly just need to wait until the components polyfills have loaded to build out the templates.

**Comment by ThatJoeMoore** [comment]
@kebowdog I've just pushed a version of every-feature.html that should work with IE11.  It's kind of hacky right now, but it should kind of work until I can make some changes to our web component polyfills that will make it better.

**Comment by kebowdog** [comment]
So in IE on every-feature.html it's slightly off-center with less than 4 columns: 
image

**Comment by ThatJoeMoore** [comment]
Yeah, I think that's related to the bugs in IE11s flexbox implementation.  There's not much I can do about that.  I think that, if the container starts with less than 4, it'll be fine, though.

**Comment by kebowdog** [comment]
Makes sense. I'm not too worried about it, just thought I'd mention it. For my part you're good to merge, unless @Gi60s or @katrialesser has any objections?

**Comment by ThatJoeMoore** [comment]
I'm going to wait to merge until @katrialesser has a chance to check it out on her calendar site (which is what kicked off all of this).

**Comment by ThatJoeMoore** [comment]
Katria has reviewed and approved this, so let's :shipit: !

**Pull request merged**